code-collab-app/src/editorpage.js

```diff
@@ -32,70 +30,68 @@ class Editorpage extends React.Component {
     if (!cellContents) {
       cellContents = '';
     }
-
-    this.setState({
-      cellText: this.state.cellText.concat(cellContents),
-      cellLocked: this.state.cellLocked.concat(false),
-    });
-
-    this.setState({
-      cellCount: this.state.cellCount + 1,
-    });
+    this.props.peer_data.add_new_cell(cellContents);
+    // TODO - maybe call peer_data method to update other peers
   }
 
   // Locks a cell so the user cannot edit it
   // key: the index of the cell to lock
   lockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = true;
-    this.setState({ cellLocked:temp });
+    // Lock cell so user may not edit
+    this.props.peer_data.set_cell_locked(key, true);
   }
 
-  // Unlocks a cell so the user may edit it
+  // Unlock a cell so the user may edit it
   // key: the index of the cell to unlock
   unlockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = false;
-    this.setState({ cellLocked:temp });
+    this.props.peer_data.set_cell_locked(key, false);
```

Comment:
I added this in my PR (https://github.com/paulchenpmc/codeCollab/pull/18/files), but before we unlock this we should update the cell contents. So that means this function should supply the updatedCellContents as an argument when called from the peer logic.

Also I'm getting confused looking at this PR. I see some elements from my PR, but it's also missing some? For example in this function in my PR there is a call to `this.setCellText(key, updatedCellContents);` but that is missing here. However, the `onBlur` event from my PR is in this.

Another example, this PR is entirely missing the `setCellText` function. I have some serious concerns about how you merged these two PRs together,

Reply:
I merged your PR first and then changed the logic so that everything is available in peer_data, which is why you see some bits but not others. As Bailey mentioned in the comment below, you can look at the last commit in this PR to get a better idea.

The reason why we don't need `updatedCellContents` in `setCellText` is because the event listener from `Textfield` now directly calls a method in peer_data everytime anything in the cell is changed. So we should already have up to date cell info in peer_data before we leave the cell.

Again, the setCellText was removed because the `TextField` now directly calls a function in peer_data.

To summarize, Editor page is now have an injected copy of peer_data store. Any changes that are made in the Editor page flow towards peer_data store and then trickle back to Editor page state via props.

Sorry, I had to do all of this because we need all the updated data in peer_data so that it can be communicated with other peers.

Again, you can look at the last commit in this peer, and ask me if anything doesn't make sense.

Reply:
Makes sense. I will resolve this thread.